src/foundation.cs

```diff
@@ -8752,7 +8753,7 @@ partial interface NSItemProvider : NSCopying {
 
 		[Mac (10,12)][Async]
 		[Export ("registerCloudKitShareWithPreparationHandler:")]
-		void RegisterCloudKitShare ([BlockCallback] Action<CloudKitRegistrationPreparationHandler> preparationHandler);
+		void RegisterCloudKitShare (CloudKitRegistrationPreparationAction preparationHandler);
```

Comment:
IMHO the main problem is that the generator defaults to `[CCallback]`, and you need it to be `[BlockCallback]`. Getting it wrong means the generated code is wrong, and any users will crash.